src/pages/Commits.tsx

```diff
@@ -40,6 +43,15 @@ export const Commits = () => {
   const dispatchToast = useToastContext();
   const { replace } = useHistory();
   const { search } = useLocation();
+  const hasSeenChartToggle = Boolean(Cookies.get(HAS_SEEN_COMMIT_CHART_TOGGLE));
+  const [isOpenChartToggle, setIsOpenChartToggle] = useState(
+    !hasSeenChartToggle
+  );
+  useEffect(() => {
+    if (!hasSeenChartToggle && !isOpenChartToggle) {
+      Cookies.set(HAS_SEEN_COMMIT_CHART_TOGGLE, "true");
+    }
+  }, [hasSeenChartToggle, isOpenChartToggle]);
```

Comment:
Since the commit chart toggle component is only used by the commit chart its state logic should in the same file. It doesn't need to be drilled through from the Commits page.

Reply:
The state is controlled in the parent component because the CommitChart will rerender everytime new data is loaded which would reset the state.